main.go

```diff
@@ -30,10 +31,15 @@ func main() {
 }
 
 func main1() int {
-	var err error
+	gunkConfig, err := config.Load("")
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error: %v\n", err)
+		return 1
+	}
 	switch kingpin.MustParse(app.Parse(os.Args[1:])) {
 	case gen.FullCommand():
-		err = generate.Generate("", *genPatterns...)
+		gunkConfig.Patterns = *genPatterns
+		err = generate.Generate(gunkConfig)
```

Comment:
I was actually imagining that each subcommand would parse the config separately. Reasons:

* Not all subcommands will need the config (think `help`, `version`, likely `gunk fmt`, etc)
* The subcommand interface should revolve around a list of arguments (strings) directly from the command line, leaving it up to the subcommand to parse flags and interpret the parameters
* Should only be four lines of simple code for each of the subcommands that need a config

I realise that the old `Generate` uses a workdir parameter instead of `os.Getwd`. That was simply so that I wouldn't have to rely on a hacky `os.Chdir` in the tests. Perhaps there's a better way to deal with that.

Reply:
Yeah, I jumped between the two. A couple of approaches I had the config being parsed from each subcommand, but I ended up going with this because I could quite see a situation where the config file parsing would be different between the subcommands. However, what you say makes sense, I'll make the change.